3_javascript/javascript2/week1/script.js

```diff
@@ -0,0 +1,128 @@
+//Find and count the Danish letters
+
+function countDanishLetters(inputString) {
+  const danishLetters = {
+    æ: 0,
+    ø: 0,
+    å: 0,
+  };
+
+  const lowercaseString = inputString.toLowerCase();
+
+  for (const char of lowercaseString) {
+    if (danishLetters.hasOwnProperty(char)) {
+      danishLetters[char]++;
+    }
+  }
+
+  const total = Object.values(danishLetters).reduce(
+    (acc, count) => acc + count,
+    0
+  );
+
+  const result = { total, ...danishLetters };
+
+  return result;
+}
+
+const danishString = "Jeg har en blå bil";
+const result1 = countDanishLetters(danishString);
+console.log(result1);
+
+const danishString2 = "Blå grød med røde bær";
+const result2 = countDanishLetters(danishString2);
+console.log(result2);
+
+//Spirit animal name generator
+
+document.addEventListener("DOMContentLoaded", function () {
+  const generateButton = document.getElementById("generateButton");
+  const spiritAnimalElement = document.getElementById("spiritAnimal");
+  const newAnimalButton = document.getElementById("newAnimalButton");
+  const eventSelector = document.getElementById("eventSelector");
+  const generateEventButton = document.getElementById("generateEventButton");
+
+  generateButton.addEventListener("click", function () {
+    generateSpiritAnimal();
+  });
+
+  newAnimalButton.addEventListener("click", function () {
+    generateSpiritAnimal();
+  });
+
+  generateEventButton.addEventListener("click", function () {
+    const selectedEvent = eventSelector.value;
+    if (selectedEvent === "click") {
+      generateSpiritAnimal();
+    }
```

Comment:
can you please explain why are you checking for the event value here?

Reply:
To be honest I asked chatGPT for some help (will try to avoid it from now on), I thought this was mandatory to check the event value. But now that you've raised the question, I see that there's no point in it. Also I realised I don't need  other `eventListeners` other than "click" ones. Thank you!

Reply:
thank you for your honesty and I am glad you noticed <3 Can I screenshot your comment to prove my point to the class? 🙏